libs/image-editor/ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/preview/PreviewImageFragment.kt

```diff
@@ -168,19 +181,27 @@ class PreviewImageFragment : Fragment() {
     }
 
     private fun setupObservers() {
-        viewModel.uiState.observe(this, Observer { state ->
-            (previewImageViewPager.adapter as PreviewImageAdapter).submitList(state.viewPagerItemsStates)
+        viewModel.uiState.observe(this, Observer { uiStateEvent ->
+            uiStateEvent?.getContentIfNotHandled()?.let { state ->
+                (previewImageViewPager.adapter as PreviewImageAdapter).submitList(state.viewPagerItemsStates)
+                cropActionMenu?.isEnabled = state.editActionsEnabled
+                UiHelpers.updateVisibility(thumbnailsTabLayout, state.thumbnailsTabLayoutVisible)
+            }
         })
 
-        /*viewModel.loadIntoFile.observe(this, Observer { fileState ->
-            if (fileState is ImageStartLoadingToFileState) {
-                loadIntoFile(fileState.imageUrl)
+        viewModel.loadIntoFile.observe(this, Observer { fileStateEvent ->
+            fileStateEvent?.getContentIfNotHandled()?.let { fileState ->
+                if (fileState is ImageStartLoadingToFileState) {
```

Comment:
I know this was here before, but it feels weird that we are propagating different states to the UI even though the UI cares only about a single state - ImageStartLoadingToFileState. I think we might want to consider separating the state from the "start loading an image into a file" action. Wdyt?

Reply:
I myself want to revisit this logic that came from single image editing and added a TODO in the view model. As this is non-blocking added a note in the project.